tests/FakeItEasy.Specs/StringExtensions.cs

```diff
@@ -0,0 +1,70 @@
+namespace FakeItEasy.Specs
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using FakeItEasy.Creation;
+    using Xbehave;
+    using Xbehave.Sdk;
+
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Creates a step builder that can be used to refer to a type.
+        /// Useful when we want to establish a condition with actually performing an action.
+        /// </summary>
+        /// <typeparam name="T">The type to look at.</typeparam>
+        /// <param name="text">A description of the type's relevant quality.</param>
+        /// <returns>A step builder.</returns>
+        [SuppressMessage("Microsoft.Design", "CA1004:GenericMethodsShouldProvideTypeParameter", Justification = "Used to indicate the type to see.")]
+        public static IStepBuilder See<T>(this string text)
+        {
+            return text.x(() => { });
+        }
+
+        /// <summary>
+        /// Creates a step builder that can be used to refer to a method that
+        /// creates a type, such as a constructor.
+        /// Useful when we want to establish a condition with actually performing an action.
+        /// </summary>
+        /// <typeparam name="T">The type to look at.</typeparam>
+        /// <param name="text">A description of the method's relevant quality.</param>
+        /// <param name="func">
+        /// The constructor (or other method that creates an instance) to look at.
+        /// Will not be executed.
+        /// </param>
+        /// <returns>A step builder.</returns>
+        [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "func", Justification = "Used to indicate the method to see.")]
+        public static IStepBuilder See<T>(this string text, Func<T> func)
+        {
+            return text.x(() => { });
+        }
+
+        /// <summary>
+        /// Creates a step builder that can be used to refer to a type's method.
+        /// Useful when we want to establish a condition with actually performing an action.
+        /// </summary>
+        /// <typeparam name="T">The type to look at.</typeparam>
+        /// <param name="text">A description of the method's relevant quality.</param>
+        /// <param name="func">The method to look at. Will not be executed.</param>
+        /// <returns>A step builder.</returns>
+        [SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "func", Justification = "Used to indicate the method to see.")]
+        public static IStepBuilder See<T>(this string text, Func<T, Action<Type, IFakeOptions>> func)

```

Comment:
Actually, it' s not that simple... Currently it's used like this:


    .See<DomainEventFakeOptionsBuilder>(_ => _.BuildOptions);

where `BuildOptions` is a method. For this to work, there must be an overload of `See` with the right signature, i.e. with a delegate parameter whose return type is a delegate that matches the method group. And we can't really make it generic (`See<T, T1, T2>(this string text, Func<T, Action<T1, T2>> func)`), because then, we would need to specify all type parameters. And anyway, we would need to make as many overloads as there are possible signatures...

So, even though it's nice to be able to use the method group like that, maybe we should do something like this instead:

    .See(nameof(DomainEventFakeOptionsBuilder.BuildOptions));